From bfa15ba6e9028e16f0bd9003303769dc78d43d0a Mon Sep 17 00:00:00 2001
From: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>
Date: Fri, 20 Oct 2017 11:05:52 +0300
Subject: [PATCH] Speed up repo synchronization

repo can separate the "network" sync (the git fetch part, network bound)
from the "local" sync (the git rebase part, compute & i/o bound).
Use this to speed up repo sync.

Signed-off-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>
---
 bitbake/lib/bb/fetch2/repo.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/bitbake/lib/bb/fetch2/repo.py b/bitbake/lib/bb/fetch2/repo.py
index 1be91cc698e3..8749f303add5 100644
--- a/bitbake/lib/bb/fetch2/repo.py
+++ b/bitbake/lib/bb/fetch2/repo.py
@@ -75,7 +75,12 @@ class Repo(FetchMethod):
             runfetchcmd("repo init -m %s -b %s -u %s://%s%s%s" % (ud.manifest, ud.branch, ud.proto, username, ud.host, ud.path), d, workdir=repodir)
 
         bb.fetch2.check_network_access(d, "repo sync %s" % ud.url, ud.url)
-        runfetchcmd("repo sync", d, workdir=repodir)
+        # FIXME: repo can separate the "network" sync (the git fetch part, network bound)
+        # from the "local" sync (the git rebase part, compute & i/o bound).
+        num_net_jobs = int(d.getVar("BB_NUMBER_THREADS") or "4")
+        num_local_jobs = int(num_net_jobs) * 3
+        runfetchcmd("repo sync -n -j%d" % num_net_jobs, d, workdir=repodir)
+        runfetchcmd("repo sync -l -j%d" % num_local_jobs, d, workdir=repodir)
 
         scmdata = ud.parm.get("scmdata", "")
         if scmdata == "keep":
-- 
2.7.4

